src/Query.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import ApolloClient, { ObservableQuery } from 'apollo-client';
+import { DocumentNode } from 'graphql';
+import { ZenObservable } from 'zen-observable-ts';
+import * as invariant from 'invariant';
+import * as pick from 'lodash.pick';
+
+import shallowEqual from './shallowEqual';
+import ApolloConsumer from './ApolloConsumer';
+
+import {
+  MutationOpts,
+  ChildProps,
+  OperationOption,
+  ComponentDecorator,
+  QueryOpts,
+  QueryProps,
+  MutationFunc,
+  OptionProps,
+} from './types';
+
+type Props = {
```

Comment:
Use an interface and it will need to be exported when `tsc -d` runs and creates the `.d.ts` file.  I usually name these after the component so they are more useful in autocomplete, such as `QueryProps` here - aids is composition of any components that might use these.